src/components/QRCode.tsx

```
@@ -0,0 +1,42 @@
+    initialAutoExecute: false,
+  });
+
+  useEffect(() => {
+    executeQRCodeGeneration(data);
+  }, [data, executeQRCodeGeneration]);
+
+  return (
+    <Box
+      component="div"
+      {...rest}
+      dangerouslySetInnerHTML={{ __html: qrCodeHTML || "" }}
```

Biome (1.9.4)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)